### PR TITLE
tfjs_bundle related improvements 

### PR DESCRIPTION
**Pull request by jinjingforever**
- Hook up the `external` parameter of the `tfjs_bundle` rule with `rollup_bundle`.
- Add the `globals` parameter to the `tfjs_bundle` rule.
- Remove the unused `kwargs` parameter from `tfjs_bundle`.
- Update the external and/or globals parameter of the `tfjs_bundle` rules in core and cpu.  
- Use bazel to build cpu backend npm.
 
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5337)

### CODE REVIEW

**Comment by mattsoulanille** [comment]
This failed due to Bazel lint errors. Run `yarn bazel:format` to fix them. I should add this to the instructions

**Comment by jinjingforever** [comment]
Ah thanks! Fixed the lint error. PTAL

**Comment by jinjingforever** [comment]
Looks like I have to temporarily disable the "test-bundle-size" test for wasm because the master branch has an error when running the test (`build-deps-ci` command/script in wasm is not found).

**Comment by mattsoulanille** [comment]
I have a fix for that: https://github.com/tensorflow/tfjs/pull/5340

**Comment by jinjingforever** [comment]
Nice! Thanks! I will merge that here after it is merged.

**Review by mattsoulanille — COMMENTED** [review]
> Reviewed 4 of 5 files at r1, 1 of 1 files at r2, 3 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5337)* status: 0 of 1 approvals obtained

___
*[tools/rollup_template.config.js, line 87 at r3](https://reviewable.io/reviews/tensorflow/tfjs/5337#-Mel_gw632wUDm4aRKpR:-Mel_gw632wUDm4aRKpS:bi13gnx) ([raw file](https://github.com/tensorflow/tfjs/blob/9eb4a9fd4b58a8025e32d8a5db8dd27013a60a9e/tools/rollup_template.config.js#L87)):*
> ```
>     banner: preamble,
> ```
Thanks!
___
*[tools/rollup_template.config.js, line 92 at r3](https://reviewable.io/reviews/tensorflow/tfjs/5337#-Mel_QM59Cp8LVB8a5hi:-Mel_QM59Cp8LVB8a5hj:b-czaegg) ([raw file](https://github.com/tensorflow/tfjs/blob/9eb4a9fd4b58a8025e32d8a5db8dd27013a60a9e/tools/rollup_template.config.js#L92)):*
> ```JavaScript
>     globals: TEMPLATE_globals,
>   },
>   external: TEMPLATE_external,
> ```

I'm surprised this isn't a native option of `rollup_bundle`, but it doesn't appear in [the docs](https://bazelbuild.github.io/rules_nodejs/Rollup.html#rollup_bundle). This looks like a good solution.

**Review by mattsoulanille — APPROVED** [review]
LGTM modulo re-enabling the wasm bundle size test! Thanks for writing these improvements. 

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5337#-:-MelbboLDASKH9vNcnLX:b-8ndkns)* status: :shipit: complete! 1 of 1 approvals obtained

**Review by jinjingforever — COMMENTED** [review]
Thank you! My latest commit has reenabled the bundle size test.

BTW I really like how the rollup config is generated. Nice approach! 

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5337#-:-Melbwg24r4OKcBEmkLh:b-x36dwv)* status: :shipit: complete! 1 of 1 approvals obtained

**Pull request merged**
